Library/Homebrew/cask/cask_loader.rb

```diff
@@ -212,8 +212,9 @@ def load(config:)
 
         json_cask.deep_symbolize_keys!
 
-        # Use the cask-source API if there are any `*flight` blocks
-        if json_cask[:artifacts].any? { |artifact| FLIGHT_STANZAS.include?(artifact.keys.first) }
+        # Use the cask-source API if there are any `*flight` blocks or the cask has multiple languages
+        if json_cask[:artifacts].any? { |artifact| FLIGHT_STANZAS.include?(artifact.keys.first) } ||
+           json_cask[:languages].any?
```

Comment:
For post-this PR: this makes me wonder if we actually want to include all artefacts and languages in the JSON. It's could to bloat it quite a bit unnecessarily?

Reply:
Yeah, it might be okay to do. Some casks like firefox have 51 languages, but most casks don't have any at all. Think it's worth doing it that way?

Reply:
@Rylan12 Perhaps just compare the size of the overall (gzipped) JSON output? It's it's noticeable then optimise, if not: leave it as-is.

Reply:
Why gzipped? We don't currently compress the JSON files before downloading, right?

I will compare later when I have some more time.

Reply:
@Rylan12 I think/thought we're downloading them gziped. If not and we can do that transparently (a non-trivial number of web servers will do this): we should.

Reply:
Yes, sorry for the delay. Let's merge and re-evaluate

Reply:
I checked: we use `--compressed` which means we will download files gzipped by the server.